File: server/ProjectPokemon/ProjectPokemon/Models/Database/Entities/Team.cs
namespace ProjectPokemon.Models.Database.Entities {
    public class Team {
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; } = null;
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public ICollection<PokemonTeam> PokemonsTeam { get; set; } = [];
    }
}

File: client/project-pokemon/src/app/models/team.ts
import { PokemonTeam } from "./pokemon-team";

export interface Team {
    id: number;
    name: string;
    description?: string | null;
    userId: number;
    pokemons: PokemonTeam[];
    isExpanded: boolean;
}

export interface GetTeamDto {
    id: number;
    name: string;
    description?: string | null;
    userId: number;
}

export interface PostTeamDto {
    name: string;
    description?: string | null;
    userId: number;
}

export const MAX_POKEMON_PER_TEAM = 6;
